### ISSUE TITLE
Many many empty files in $HOME/.config/matplotlib

### ISSUE THREAD

**Issue by rainwoodman** [matplotlib/matplotlib#16104]
### Bug report

**More than 30,000 files in $HOME/.config/matplotlib**

When doing a clean house in my home directory, I found $HOME/.config/matplotlib has about 30,000 files named like tmpXXXXXXX. A few files I looked were empty. The directory is so large that I didn't give it enough time for ls -l to finish.

I have been using this home directory for many years. Therefore I do not know if there were produced from a recent version of matplotlib, or due to a historical glitch. It looks strange that matploblib puts any temporary files in $HOME.

This could be a configuration issue -- any hints on what is going on?

**Comment by tacaswell** [comment]
That is very concerning and nothing jumps to mind right away.

If you look at the timestamps on a few selected ones can you get a sense of how old they might be?  "many years" can cover many Matplotlib versions.  The next thing might be to move the directory full of files to a different path and see if we are creating new ones again.

**Comment by timhoffm** [comment]
Also,

- Do you remember any crashes or hard terminations when working in the console with matplotlib or a GUI program that uses Matplotlib? 
- Do you use animations?

**Comment by rainwoodman** [comment]
Hard crashes are frequent (10%). I run with large MPI jobs (4K processes). Alternatively sometimes the usage is via jupyter notebook. I do not use animations.

I browsed through the current version of the config / cache dir handling and couldn't see anything that could mix them up internally from matplotlib. Unfortunately I have purged the dir (I was tracing down some unexpected slowdown).
I haven't seen new entries created in that directory yet. I will observe a while and report back.
